modules/programs/ne.nix

```
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.ne;
+
+  autoPrefFiles = let
+    autoprefs = if cfg.defaultPreferences != "" then
+      cfg.automaticPreferences // {
+        ".default" = cfg.defaultPreferences;
+      } # include defaultPreferences as .default#ap
+    else
+      cfg.automaticPreferences;
```

A bit more idiomatic:
```suggestion
    autoprefs =
      cfg.automaticPreferences
      // optionalAttrs (cfg.defaultPreferences != "") {
        ".default" = cfg.defaultPreferences;
      }
```